x/tokenfactory/keeper/msg_server.go

```diff
@@ -0,0 +1,159 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/x/tokenfactory/types"
+)
+
+type msgServer struct {
+	Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
+
+var _ types.MsgServer = msgServer{}
+
+func (server msgServer) CreateDenom(goCtx context.Context, msg *types.MsgCreateDenom) (*types.MsgCreateDenomResponse, error) {
```

Comment:
(Doesn't block merge)

Should we add a governance tunable osmo cost for minting denoms?

Reply:
I thought that we should add it for spam resistance, but then realized that we already have a non-spam resistant way of creating denoms... the ibc-transfer module.

Although arguably, that should also require a fee perhaps.

Reply:
Yeah, I'd really prefer to add it in. (I think we should add it in there once we figure out a sensible way for there to be a fee!) 

IBC has a notable annoyance overhead, this will be trivial spam.